Formula/boost-python3.rb

```diff
@@ -0,0 +1,85 @@
+class BoostPython3 < Formula
+  desc "C++ library for C++/Python 3 interoperability"
+  homepage "https://www.boost.org/"
+  url "https://dl.bintray.com/boostorg/release/1.66.0/source/boost_1_66_0.tar.bz2"
+  sha256 "5721818253e6a0989583192f96782c4a98eb6204965316df9f5ad75819225ca9"
+  head "https://github.com/boostorg/boost.git"
+
+  bottle do
+    cellar :any
+    sha256 "7eb61418c5c91034a7bc00cd5b56070a901afa4adfd3a7c0551e3809da4673a1" => :high_sierra
+    sha256 "1821cfeabaeafe8ab6e003a8715fdff1c481afee0ebdef423082a321bc24d505" => :sierra
+    sha256 "0d49ef79a0be83226cdb0fb2b010f6bcc867ce343fa980aee2ddb6356327688f" => :el_capitan
+  end
+
+  depends_on "python3"
+  depends_on "boost"
+
+  needs :cxx11
+
+  def install
+    # "layout" should be synchronized with boost
+    args = ["--prefix=#{prefix}",
+            "--libdir=#{lib}",
+            "-d2",
+            "-j#{ENV.make_jobs}",
+            "--layout=tagged",
+            "--user-config=user-config.jam",
+            "threading=multi,single",
+            "link=shared,static"]
+
+    # Trunk starts using "clang++ -x c" to select C compiler which breaks C++11
+    # handling using ENV.cxx11. Using "cxxflags" and "linkflags" still works.
+    args << "cxxflags=-std=c++11"
+    if ENV.compiler == :clang
+      args << "cxxflags=-stdlib=libc++" << "linkflags=-stdlib=libc++"
+    end
+
+    # disable python detection in bootstrap.sh; it guesses the wrong include directory
+    # for Python 3 headers, so we configure python manually in user-config.jam below.
+    inreplace "bootstrap.sh", "using python", "#using python"
+
+    pyver = Language::Python.major_minor_version "python3"
+
+    py_prefix = `python3 -c "import sys; print(sys.prefix)"`.strip
```

Comment:
We don't need to do this any more, IMO, for `python3`. Now we're _only ever_ using Homebrew's `python3` we _know_ what these variables will be. This one for example is simply:

```ruby
Formula["python3"].opt_frameworks/"Versions/#{pyver}"
```

Reply:
Should we also *enforce* using Homebrew's python in `boost-python`? Currently it takes the `python` which is in your $PATH.